lib/views/components/content/user_avatar/user_avatar.dart

```diff
@@ -25,35 +28,53 @@ class UserAvatar extends StatelessWidget {
   // working and displays the user's profile picture behind the InkWell response.
 
   @override
-  Widget build(BuildContext context) {
-    return CircleAvatar(
-      radius: radius,
-      backgroundColor: DynamicUserAvatarViewModel.centauriToColor(
-        details.userCentauriPoints,
-        Theme.of(context).brightness,
-      ),
-      child: Stack(
-        children: [
-          Center(
-            child: Text(
-              key: initialDisplayNameKey,
-              // Display the first letter of the [displayName] parameter (user's initial).
-              // If the display name is empty, display an empty string (because
-              // it causes an error if it is empty).
-              details.displayName.isEmpty
-                  ? ""
-                  : details.displayName.substring(0, 1),
-            ),
+  Widget build(BuildContext context, WidgetRef ref) {
+    final userCentauriAsync =
+        ref.watch(userCentauriPointsViewModelProvider(details.userID));
+
+    final content = Stack(
+      children: [
+        Center(
+          child: Text(
+            key: initialDisplayNameKey,
+            // Display the first letter of the [displayName] parameter (user's initial).
+            // If the display name is empty, display an empty string (because
+            // it causes an error if it is empty).
+            details.displayName.isEmpty
+                ? ""
+                : details.displayName.substring(0, 1),
           ),
-          Material(
-            shape: const CircleBorder(),
-            clipBehavior: Clip.hardEdge,
-            color: Colors.transparent,
-            child: InkWell(
-              onTap: onTap,
-            ),
+        ),
+        Material(
+          shape: const CircleBorder(),
+          clipBehavior: Clip.hardEdge,
+          color: Colors.transparent,
+          child: InkWell(
+            onTap: onTap,
           ),
-        ],
+        ),
+      ],
+    );
+
+    return userCentauriAsync.when(
+      data: (userCentauri) {
+        final backgroundColor = centauriToUserAvatarColor(
+          userCentauri,
+          Theme.of(context).brightness,
+        );
+        return CircleAvatar(
+          radius: radius,
+          backgroundColor: backgroundColor,
+          child: content,
+        );
+      },
+      loading: () => CircleAvatar(
+        radius: radius,
+        backgroundColor: loadingUserAvatarColor,
+        child: content,
+      ),
+      error: (error, _) => throw Exception(
+        "${CircularValue.debugErrorTag} User avatar color error: $error",
```

Comment:
You can make this code much simpler by taking advantage of `AsyncValue::value` which returns the previous value if it exists or null otherwise, and re-throws the error if the `AsyncValue` is in an error state; and with the fact that `centauriToUserAvatarColor` returns the loading value when the number of points given in argument is `null`.
```
    final backgroundColor = centauriToUserAvatarColor(
      userCentauriAsync.value,
      Theme.of(context).brightness,
    );

    return CircleAvatar(
      radius: radius,
      backgroundColor: backgroundColor,
      child: content,
    );
```

In fact, using this method, you can go back to the way this file originally was by adding very few lines.

Reply:
You are right.

Fixed in: [ad5cdda](https://github.com/ProximaEPFL/proxima/pull/348/commits/ad5cddab335c641e4c227995e7a579559f0aaf19)

Thank you 💯

Reply:
Thanks. However, why did you choose to keep the exception re-throw? I feel like it does not bring much to the code, but still increases slightly the cognitive complexity while destroying the stack trace of the original error (preventing us from knowing from where it originated).

Reply:
Oh, I see what you mean 😄 . Moreover since we don't always have a circular value backing the the widget, is the not relevant and this is better to directly throw the error without breaking the stack trace.

Removed in: [7850437](https://github.com/ProximaEPFL/proxima/pull/348/commits/7850437df297d31ee796897e115b38325b7682e3)

Thank you 💯